helpers/functions.php

```diff
@@ -7,19 +7,24 @@
 use LaravelLang\Config\Facades\Config;
 
 if (! function_exists('localizedRoute')) {
-    function localizedRoute(string $route, array $parameters = [], bool $absolute = true): string
-    {
-        $locale = Config::shared()->routes->names->parameter;
-        $prefix = Config::shared()->routes->namePrefix;
+    function localizedRoute(
+        string $route,
+        array $parameters = [],
+        bool $absolute = true,
+        bool $redirectDefault = false
+    ): string {
+        $currentLocale = app()->getLocale();
+        $defaultLocale = config('app.locale');
```

Comment:
Perhaps you didn't know, but:

```php
public function getLocale()
{
    return $this['config']->get('app.locale');
}
```
https://github.com/laravel/framework/blob/12.x/src/Illuminate/Foundation/Application.php#L1545-L1553

So `$currentLocale` will always be equal to `$defaultLocale`.

In this case, it is better to use a comparison with fallback localization.

```php
use LaravelLang\Locales\Facades\Locales;

$fallback = Locales::raw()->getFallback();
```